components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -14723,4 +14724,87 @@ public String getUserID(String userName) throws APIManagementException {
         }
         return list;
     }
+
+    /**
+     * Add a new endpoint registry
+     *
+     * @param endpointRegistry EndpointRegistry
+     * @param tenantID  ID of the owner's tenant
+     * @return registryId
+     */
+    public String addEndpointRegistry(EndpointRegistry endpointRegistry, int tenantID) throws APIManagementException {
+        String query = SQLConstants.ADD_ENDPOINT_REGISTRY_SQL;
+        String uuid = UUID.randomUUID().toString();
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement ps = connection.prepareStatement(query)) {
+            ps.setString(1, uuid);
+            ps.setString(2, endpointRegistry.getName());
+            ps.setString(3, endpointRegistry.getType());
+            ps.setString(4, endpointRegistry.getMode());
+            ps.setInt(5, tenantID);
+            ps.setString(6, endpointRegistry.getOwner());
+            ps.setString(7, "");
+            ps.setString(8, "");
+            ps.executeUpdate();
+        } catch (SQLException e) {
+            handleException("Error while adding new endpoint registry: " + endpointRegistry.getName(), e);
+        }
+        return uuid;
+    }
+
+    /**
+     * Return the details of an Endpoint Registry
+     *
+     * @param regsitryId Endpoint Registry Identifier
+     * @return Endpoint Registry Object
+     * @throws APIManagementException
+     */
+    public EndpointRegistry getEndpointRegistryByUUID(String regsitryId) throws APIManagementException {
+        String query = SQLConstants.GET_ENDPOINT_REGISTRY_BY_UUID;
+        EndpointRegistry endpointRegistry = new EndpointRegistry();
```

Comment:
Shouldn't the instantiation of the EndpointRegistry go inside  the **f (rs.next()) {** block ? otherwise not found would not return null.

Reply:
Fixed